src/IO/S3Common.cpp

```diff
@@ -677,6 +677,12 @@ namespace S3
     ClientFactory::ClientFactory()
     {
         aws_options = Aws::SDKOptions{};
+
+        /// We don't load EC2 metadata.
```

Comment:
Why? It's one of the most popular ways to get auth info. Some cloud providers for clickhouse use it.

Reply:
Here I'm just keeping EC2 metadata disabled as it was before my PR. See https://github.com/ClickHouse/aws-sdk-cpp/commit/4bd1ca4af129d22ea7d53b8148520e08bd77de32. I just slightly changed the way how it's implemented in the code.

Why it was disabled by us in https://github.com/ClickHouse/aws-sdk-cpp/commit/4bd1ca4af129d22ea7d53b8148520e08bd77de32:
To create `S3::PocoHTTPClientConfiguration` we use `S3::ClientFactory::instance().createClientConfiguration()` and `S3::ClientFactory::instance()` calls `S3::ClientFactory::ClientFactory()` which calls `Aws::InitAPI()` which calls `Aws::Internal::InitEC2MetadataClient()`.

And  `Aws::Internal::InitEC2MetadataClient()` uses `Aws::Client::ClientConfiguration` instead of our subclass `S3::PocoHTTPClientConfiguration`. However our function `PocoHTTPClientFactory::CreateHttpClient()` assumes that the passed `clientConfiguration` is a `S3::PocoHTTPClientConfiguration` and crashes.

To fix that we could pass our `S3::PocoHTTPClientConfiguration` to `Aws::InitAPI()` and then to `Aws::Internal::InitEC2MetadataClient()`, but if we do so we will have circular dependency `S3::PocoHTTPClientConfiguration` -> `S3::ClientFactory` -> `Aws::InitAPI()` -> `S3::PocoClientConfiguration`, and it's not very trivial how to fix that.

Anyway I'm not doing things worse here as I said. I suppose we can try to resolve that circular dependency in a separate PR.

Reply:
I'll investigate that further.

Reply:
We load EC2 metadata by ourselves so it should be disabled in aws sdk core, but not by changing that environment variable. I changed the code and wrote a comment about it - see https://github.com/ClickHouse/ClickHouse/pull/43020/commits/9014be89681c4509485e7e809fdda27ccef817d6